### DISCUSSION TITLE
Apple Vision Pro - can we develop Air applications that make use of its features?

### DISCUSSION THREAD

**Discussion by NikosLaskaridis** [airsdk/Adobe-Runtime-Support#2937] [General]
Can we develop Air apps targeting Apple Vision Pro? Like an AR app that combines camera and location info?

**Comment by ajwfrost** [comment]
Currently it looks like existing applications, including AIR ones, can probably be run on the Vision hardware, but to create an app which is more suited and designed for this, it would need to be set up with the appropriate UI frameworks and mechanisms .. which aren't how AIR does things, sadly. The iOS runtime integration was written quite a while ago and is heavily Objective-C based, whereas for Vision Pro development it needs to be Swift. Could be a good time for us to make some updates, or we could look into an alternative mechanism that just allows SWF content to be added in to existing Vision apps/windows.

So, it's something we'll keep an eye on, but we'd also want to get some indication of demand for properly supporting this product..

thanks

**Reply by NikosLaskaridis** [reply]
Thanks for answering and looking it up Andrew.
